yt_idv/scene_data/block_collection.py

```diff
@@ -351,3 +380,46 @@ def _block_collection_outlines(
     data_rendering.display_name = display_name
 
     return data_collection, data_rendering
+
+
+class _DummyBlock:
+    def __init__(self, grid) -> None:
+        self.grid = grid
+        self.LeftEdge = grid.LeftEdge
+        self.RightEdge = grid.RightEdge
+
+    @property
+    def source_mask(self):
+        return np.ones(self.grid.ActiveDimensions, dtype="uint8")
+
+
+class GridCollection(AbstractDataCollection):
+    name = "block_collection"
+    data_source = traitlets.List(trait=traitlets.Instance(YTDataContainer))
+
+    _field = None
+    _no_ghost = None
+
+    def _initialize_data_source(self, field, no_ghost=False):
+        self._field = field
+        self._no_ghost = no_ghost
+        self._yt_geom_str = str(self.data_source[0].ds.geometry)
+
+    def _volume_iterator(self):
+        for grid in self.data_source:
+            yield _DummyBlock(grid)
+
+    def _get_volume_data(self, block):
+        return block.grid[self._field]
+
+    def _get_ds(self):
+        return self.data_source[0].ds
+
+    def viewpoint_iter(self, camera):
+        # neglecting camera
+        for vbo_i in range(len(self.blocks)):
+            yield (vbo_i, self.texture_objects[vbo_i], self.bitmap_objects[vbo_i])
```

Comment:
So this i the one bit I couldn't figure out easily: this will just return all the grid objects without any along-ray sorting... which is fine for max intensity (and projective?) projections. but will result in incorrect results for a full transfer function.